controllers/PageController.php

```diff
@@ -457,4 +459,121 @@ public function actionToggleNumbering(int $id)
+        $mergedRevision->save();
+
+        return $this->redirect(Url::toWiki($page));
+    }
+
+    public function actionCreateCopy(int $id) {
+
+        $userIdentity = Yii::$app->user->identity->username;
+        $dateTime = new DateTime();
+
+        $page = $this->getWikiPage($id);
+        if (!$page) {
+            throw new HttpException(404, 'Wiki page not found!');
+        }
+
+        $parentId = $page->parent_page_id;
+
+        $form = (new PageEditForm(['container' => $this->contentContainer]))->forPage($id);
+        if (!$form->load(Yii::$app->request->post())) {
+            throw new HttpException(404);
+        }
+
+        $childPage = new WikiPage();
+        $childPage->title = $page->title.' conflicting copy of '. $userIdentity.' from '. $dateTime->format('Y-m-d H:i:s');
+        $childPage->parent_page_id = $parentId;
+        $childPage->content->contentcontainer_id= $page->content->contentcontainer_id;
+
+        if (!$childPage->save()) {
+            throw new HttpException(500, 'Failed to create the child page!');
+        }
+
+        if (!$childPage->id) {
+            throw new HttpException('Child page ID is not available after saving.');
+        }
+
+        $revision = new WikiPageRevision();
+        $revision->content = $form->revision->content;
+        $revision->wiki_page_id = $childPage->id;
+        $revision->revision = 1;
+        $revision->user_id = Yii::$app->user->id;
+
+        if (!$revision->save()) {
+            throw new HttpException('Failed to add content to the child page.');
+        }
+
+        return $this->redirect(Url::toWiki($page));
+    }
+
+    public function actionEditingStatus(int $id)
+    {
+        $page = $this->getWikiPage($id);
+        if (!$page) {
+            throw new HttpException(404, 'Wiki page not found!');
+        }
+
+        $user = Yii::$app->user->identity->username;
+
+        return $this->asJson([
+            'success' => true,
+            'isEditing' => $page->isEditing(),
+            'user' => $page->is_currently_editing,
+        ]); 
+    }
+
+    public function actionEditingTimerUpdate(int $id = null)
+    {   
+        $unAuthorizedLogin = false;
+        $page = $this->getWikiPage($id);
+        if (!$page) {
+            return $this->asJson([
+                'sucess' => false,
+            ]);
+        }
+
+        $user = Yii::$app->user->identity->username;
+
+        if ($page->is_currently_editing == NULL) {
+            $page->updateIsEditing();
+        }
+
+        if ($page->is_currently_editing == $user) {
+            $page->updateEditingTime();
+        }
+        else {
+            $unAuthorizedLogin = true;
+        }
+
+        return $this->asJson([
+            'success' => true,
+            'user' => $page->is_currently_editing,
+            'unAuthorizedLogin' => $unAuthorizedLogin,
```

Comment:
Is the naming `unAuthorizedLogin` really correct?
As I understand it, this only means that the user edits the page despite the warning, right?
If so, please rename it, e.g. conflictingEditing.